src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -2873,6 +2873,365 @@ static void Main()
             Diagnostic(ErrorCode.ERR_BadArgType, "x").WithArguments("1", "in string", "ref readonly int").WithLocation(9, 14));
     }
 
+    [Theory, CombinatorialData]
+    public void DefaultParameterValue_EqualsValue(bool fromMetadata)
+    {
+        var source1 = """
+            public class C
+            {
+                public static void M(ref readonly int i = 1) => System.Console.Write(i);
+            }
+            """;
+        var warning1 =
+            // (3,47): warning CS9521: A default value is specified for 'ref readonly' parameter 'i', but 'ref readonly' should be used only for references. Consider declaring the parameter as 'in'.
+            //     public static void M(ref readonly int i = 1) => System.Console.Write(i);
+            Diagnostic(ErrorCode.WRN_RefReadonlyParameterDefaultValue, "1").WithArguments("i").WithLocation(3, 47);
+        var source2 = """
+            class D
+            {
+                static void Main()
+                {
+                    int x = 2;
+                    C.M();
+                    C.M(x);
+                    C.M(ref x);
+                    C.M(in x);
+                }
+                static void M2() => C.M();
+            }
+            """;
+        var warning2 =
+            // (7,13): warning CS9503: Argument 1 should be passed with 'ref' or 'in' keyword
+            //         C.M(x);
+            Diagnostic(ErrorCode.WRN_ArgExpectedRefOrIn, "x").WithArguments("1").WithLocation(7, 13);
+        var comp = fromMetadata
+            ? CreateCompilation(source2, new[] { CreateCompilation(source1).VerifyDiagnostics(warning1).EmitToImageReference() }, options: TestOptions.ReleaseExe).VerifyDiagnostics(warning2)
+            : CreateCompilation(new[] { source1, source2 }, options: TestOptions.ReleaseExe).VerifyDiagnostics(warning1, warning2);
```

Comment:
>VerifyDiagnostics(warning1, warning2) [](http://example.com/codeflow?start=93&length=37)

I think that we should verify diagnostics on ```CompileAndVerify``` instead. The set of diagnostics will be conditional on ```fromMetadata```, but we want to verify complete set of diagnostics, including from emit phase for the compilation.